include/copc-lib/io/reader.hpp

```diff
@@ -41,9 +42,13 @@ class Reader : public BaseIO
     las::Points GetAllPoints();
 
     // Spatial query functions
-    std::vector<Node> GetNodesWithinBox(const Box &box);
-    std::vector<Node> GetNodesIntersectBox(const Box &box);
-    las::Points GetPointsWithinBox(const Box &box);
+    // Definitions taken from https://shapely.readthedocs.io/en/stable/manual.html#binary-predicates
+    std::vector<Node> GetNodesWithinBox(const Box &box, double min_resolution = std::numeric_limits<double>::min());
```

Comment:
min_res can be 0 I think

Reply:
No or it will error out. Resolution query can't be 0 at the moment, should I handle the 0 query differently? Like if it's 0 then return all nodes instead of erroring out?

Reply:
Main issue by doing that is how to handle 
`int32_t Reader::GetDepthAtResolution(double resolution) const
{
    if (resolution <= 0)
        throw std::runtime_error("Reader::GetDepthAtResolution: Query resolution must be greater than 0.");
    if (GetCopcHeader().span <= 0)
        throw std::runtime_error("Reader::GetDepthAtResolution: Octree span must be greater than 0.");

    auto root_resolution = (GetLasHeader().max.x - GetLasHeader().min.x) / GetCopcHeader().span;
    return static_cast<int32_t>(std::floor(std::max(0.0, std::log(root_resolution / resolution) / std::log(2))));
}`

Reply:
For resolution=0 in queries, it should return all nodes. For resolution=0 in GetDepth, it should return the maximum depth level.